tests/jax/utils.py

```diff
@@ -943,7 +971,14 @@ def __post_init__(self):
 
     @nn.compact
     def __call__(self, inputs, encoder_mask=None, deterministic=False):
-        del self.self_attn_mask_type  # dummy, just align to TE's impl
+        # Currently cuDNN backend only supports SWA for causal/padding_causal, follow this
+        if self.self_attn_mask_type in ["causal", "padding_causal"] and self.window_size[0] > 0:
```

Comment:
What is the expected behavior if self.self_attn_mask_type == 'padding' and self.window_size[0] > 0?

Reply:
In test_fused_attn.py, if we use padding mask and a `window_size[0] > 0`, no supported C++ backend is available, and the tests are all skipped. The EncoderLayer and DecoderLayer classes in this `utils.py` will only be used in `test_layer.py`. Added some code in `test_layer.py` to skip if `window_size[0] > 0` and not using causal/padding_causal mask.

Reply:
The logical between `test_layer.py` and `utils.py` should be seperated. We don't know when will other people make changes to `test_layer.py` or `utils.py` standalone. If someone don't know the relationship between them, they might only change only one side and spend lots of time on debugging. 

Currently, I noticed that this call doesn’t handle self.self_attn_mask_type == 'padding', nor does it provide any warning or error. It’s important not to assume this function will only be used by `test_layer.py` or `padding` will not be passed, as new developers may not be aware of these assumptions. If padding is not supported at the moment, it would be best to raise an exception within this function to prevent unintended behavior.

Reply:
Does the following code look good?

```python
if self.window_size[0] > 0:
    if self.self_attn_mask_type in ["causal", "padding_causal"]:
        encoder_mask = apply_swa_mask(
            self.self_attn_mask_type,
            encoder_mask,
            self.window_size,
        )
    else:
        raise NotImplementedError("cuDNN only supports SWA for causal and padding_causal")
```

Reply:
Looks good